src/repo.rs

```diff
@@ -802,6 +802,19 @@ impl Repository {
         }
     }
 
+    /// Create stream to write blob
+    pub fn blob_writer(&self, hintpath: Option<&Path>) -> Result<BlobWriter, Error> {
+        let path = match hintpath {
+            Some(path) => try!(path.into_c_string()).into_raw(),
```

Comment:
Yes, but you'll want to create the `CString`, store it on the stack, then call `as_ptr` later